lib/vulcan.js

```diff
@@ -244,13 +244,19 @@ Vulcan.prototype = {
           if (end === -1) {
```

Comment:
I'd like to see this broken out into a function along these lines: https://gist.github.com/garlicnation/e6137465ecd2ba9aa022